src/mono/mono/mini/aot-compiler.c

```diff
@@ -14455,6 +14472,10 @@ aot_assembly (MonoAssembly *ass, guint32 jit_opts, MonoAotOptions *aot_options)
 
 	current_acfg = NULL;
 
+	if (acfg->aot_opts.compiled_methods_outpath) {
+		fclose (acfg->compiled_methods_outfile);
```

Comment:
If dedup is enabled, `aot_assembly` could return before `fclose` and leave file descriptor dangling. Not sure if it is the only case, there might be more.

Reply:
That's a good point. I added the check when open the file. if `acfg->dedup_collect_only` is true, aot won't print out compiled method list.